README.md

```diff
@@ -51,63 +51,213 @@ conda activate lasaft
 conda install pytorch=1.7.1 cudatoolkit=11.0 -c pytorch
 conda install -c conda-forge ffmpeg librosa=0.6
 conda install -c anaconda jupyter
-pip install musdb==0.3.1 museval==0.3.0 pytorch_lightning==1.1.6 wandb==0.10.15 pydub==0.24.1 wget
+pip install musdb==0.3.1 museval==0.3.0 pytorch_lightning==1.1.6 wandb==0.10.15 pydub==0.24.1 wget hydra-core==1.1.0
 ```
 
 ### 2. Dataset: Musdb18
 
 LaSAFT was trained/evaluated on the [Musdb18](https://sigsep.github.io/datasets/musdb.html) dataset.
 
-We provide [wrapper packages](https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/tree/main/lasaft/data/musdb_wrapper) to efficiently load musdb18 tracks as pytorch tensors.
+We provide [wrapper packages](https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/tree/main/lasaft/data/musdb_wrapper.py) to efficiently load musdb18 tracks as pytorch tensors.
 
 You can also find [useful scripts](https://github.com/ws-choi/Conditioned-Source-Separation-LaSAFT/tree/main/lasaft/data) for downloading and preprocessing Musdb18 (or its 7s-samples).
 
 ### 3. Training
 
 - Below is an example to train a U-Net with LaSAFT+GPoCM, whose hyper-parameters are set as default.
     ```shell script
-    python main.py --problem_name conditioned_separation --mode train --run_id lasaft_net --musdb_root etc/musdb18_dev_wav --gpus 1 --precision 16 --batch_size 6 --num_workers 0 --pin_memory True --save_top_k 3 --save_weights_only True --patience 10 --lr 0.001 --model CUNET_TFC_GPoCM_LaSAFT
+    python train.py trainer.gpus=1 dataset.batch_size=6
     ```
-- main.py includes training scripts for several models described in the paper [1].
+- train.py includes training scripts for several models described in the paper [1].
     - It provides several options, including pytorch-lightning parameters
+    - `model/conditioned_separation: CUNET_TFC_FiLM, CUNET_TFC_FiLM_LaSAFT, CUNET_TFC_FiLM_TDF, CUNET_TFC_GPoCM, CUNET_TFC_GPoCM_LaSAFT, CUNET_TFC_GPoCM_LightSAFT, CUNET_TFC_GPoCM_TDF, default, lasaft_net, lightsaft_net`
 
 #### Examples
 
 - Table 1 in [1]
 
     - FiLM CUNet
         ```shell script
-        python main.py --problem_name conditioned_separation --mode train --musdb_root etc/musdb18_dev_wav --gpus 1 --precision 16 --batch_size 8 --num_workers 0 --pin_memory True --save_top_k 3 --save_weights_only True --patience 10 --lr 0.001 --deterministic --model CUNET_TFC_FiLM --log False
+        python train.py model=conditioned_separation/CUNET_TFC_FiLM trainer.gpus=1 training.log=False
```

Comment:
The same question,

- but in this case I prefer the full explicit configuration for paper [1] readers.
  - since each of them reproduces the result of [1]
- if you could provide an example for this case, then I will update the others.

[1] Choi, Woosung, et al. "LaSAFT: Latent Source Attentive Frequency Transformation for Conditioned Source Separation." ICASSP 2021-2021 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP). IEEE, 2021.

Reply:
Sorry I found them below

Reply:
I see what you mean. I will modify the shell script to show the hyperparameters written in the paper.